cmd/minishift/cmd/start_preflight.go

```diff
@@ -0,0 +1,331 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"github.com/docker/machine/libmachine/drivers"
+	startFlags "github.com/minishift/minishift/cmd/minishift/cmd/config"
+	miniutil "github.com/minishift/minishift/pkg/minishift/util"
+
+	"github.com/minishift/minishift/pkg/util/os/atexit"
+	"github.com/spf13/viper"
+)
+
+const (
+	StorageDisk = "/mnt/sda1"
+)
+
+// preflightChecksAfterStartingHost is executed before the startHost function.
+func preflightChecksBeforeStartingHost() {
+	switch viper.GetString(startFlags.VmDriver.Name) {
+	case "xhyve":
+		preflightCheckSucceedsOrFails(
+			startFlags.SkipCheckXHyveDriver.Name,
+			checkXhyveDriver,
+			"Checking if xhyve driver is installed",
+			false, startFlags.WarnCheckXHyveDriver.Name,
+			"See the 'Setting Up the Driver Plug-in' topic for more information")
+	case "kvm":
+		preflightCheckSucceedsOrFails(
+			startFlags.SkipCheckKVMDriver.Name,
+			checkKvmDriver,
+			"Checking if KVM driver is installed",
+			false, startFlags.WarnCheckXHyveDriver.Name,
+			"See the 'Setting Up the Driver Plug-in' topic for more information")
+	case "hyperv":
+		preflightCheckSucceedsOrFails(
+			startFlags.SkipCheckHyperVDriver.Name,
+			checkHypervDriver,
+			"Checking if Hyper-V driver is configured",
+			false, startFlags.WarnCheckHyperVDriver.Name,
+			"Hyper-V virtual switch is not set")
+	}
+}
+
+// preflightChecksAfterStartingHost is executed after the startHost function.
+func preflightChecksAfterStartingHost(driver drivers.Driver) {
+	preflightCheckSucceedsOrFailsWithDriver(
```

Comment:
It is the "pre-flight check" that "succeeds or fails"... not the "pre-flight", as that refers to a grouping, and "check" is here used as a noun, not the "verb" form, else I would have preferred to use the "gerund" form.

Kept this way to indicate in the `start_preflight.go` that two groupings of test exists, before and after the start of the VM. It is called StartingHost to be indicative to how this is named in the `start.go` file.